progress/upload_status.py

```diff
@@ -16,27 +49,66 @@ def get_directory_status(directory, sample_sheet):
     result = DirectoryStatus(directory)
 
     if not os.access(directory, os.W_OK):
-        result.status = 'invalid'
+        result.status = DIRECTORY_STATUS_INVALID
         result.message = 'Directory cannot be accessed. Please check permissions'
         return result
 
     file_list = next(os.walk(directory))[2]  # Gets the list of files in the directory
     if sample_sheet not in file_list:
-        result.status = 'invalid'
+        result.status = DIRECTORY_STATUS_INVALID
         result.message = 'Directory has no valid sample sheet file with the name {}'.format(sample_sheet)
         return result
 
-    if '.miseqUploaderInfo' not in file_list:  # no .miseqUploaderInfo file yet, has not been uploaded
-        result.status = 'new'
+    if STATUS_FILE_NAME not in file_list:  # no irida_uploader_status.info file yet, has not been uploaded
+        result.status = DIRECTORY_STATUS_NEW
         return result
 
     # Must check status of upload to determine if upload is completed
-    uploader_info_file = os.path.join(directory, '.miseqUploaderInfo')
+    uploader_info_file = os.path.join(directory, STATUS_FILE_NAME)
     with open(uploader_info_file, "rb") as reader:
         info_file = json.load(reader)
-        complete = info_file["Upload Status"] == "Complete"  # if True, done uploading, if False, partially done
-        if complete:
-            result.status = 'complete'
-        else:
-            result.status = 'partial'
-        return result
+        status = info_file[STATUS_FIELD]
+        if status in DIRECTORY_STATUS_LIST:
+            result.status = status
+        else:  # the status found in the file is not in the defined list
+            raise exceptions.DirectoryError("Cannot access directory", directory)
+
+    return result
+
+
+def write_directory_status(directory, status, run_id=None):
+    """
+    Writes a status to the status file:
+    Overwrites anything that is in the file
+
+    Writes a timestamp to the time of last written
+
+    :param directory: directory status file is in (or will be created in)
+    :param status: status to set the run to
+        Should be one of defined module level status constants
+    :return: None
+    """
+
+    if not os.access(directory, os.W_OK):  # Cannot access upload directory
+        raise exceptions.DirectoryError("Cannot access directory", directory)
+
+    uploader_info_file = os.path.join(directory, STATUS_FILE_NAME)
+    if run_id:
+        json_data = {STATUS_FIELD: status,
+                     DATE_TIME_FIELD: _get_date_time_field(),
+                     RUN_ID_FIELD: run_id,
+                     IRIDA_INSTANCE_FIELD: config.read_config_option('base_url')}
+    else:
+        json_data = {STATUS_FIELD: status,
+                     DATE_TIME_FIELD: _get_date_time_field()}
+
+    with open(uploader_info_file, "w") as json_file:
+        json.dump(json_data, json_file, indent=4, sort_keys=True)
```

Comment:
Could you also print a trailing newline to the `json_file`?  When I run `cat irida_uploader_status.info` it's printing some of the file contents on the same line as the terminal prompt string.

Reply:
Good catch, added.